code/modules/mob/dead/observer/observer.dm

```diff
@@ -477,6 +478,9 @@ Works together with spawning an observer, noted above.
 		ghost.mind.player_entity.update_panel_data(round_statistics)
 		ghost.mind.original = src
 
+	// If we're ghosting as part of previous mob deletion, do not keep refs
+	if(QDELETED(src) && mind?.original == src)
+		mind.original = null
```

Comment:
I'm inclined to think this would be better to do in just Destroy rather than if we're calling ghostize.

Reply:
It would be better ideally, but this is on purpose: there is actually handling for it in `/mob/Destroy` but by design it can only take effect if ghostize failed. If ghostize succeeds we are left with two choices: this, or nulling it BEFORE ghostize and hoping it doesn't break anything.

Reply:
I guess I could refactor ghostize to return success and act upon it instead... Thoughts?

Reply:
Up to you, I just figure assignment to null can just be done always in Destroy (namely when it goes through mob/Destroy) rather than conditionally here.

Reply:
This is existing code:
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/277ccd5e-56bb-4c17-a4f9-7404e2bedf69)

So to me it seems redundant to be nulling mind here when in a few lines it will be nulled?

Reply:
Well that's what i was explaining above, I don't understand what you mean

If ghostize didn't fail then the non-ghost mob is still stored as mind.original, hence it needing to be cleared

If ghostize fails, both mind.current and mind.original are still set and need to be cleared in that snippet

ghostize cannot know at present whether it should or not clear mind.original without adding eg. a variable passed to it, which i figured was essentially simpler to just use QDELETED

we cannot clear mind in Destroy() because it is still valid and needed for the purpose of ghostize

Reply:
There is nothing done about a "ghostize" failure, but this is also extremely minor.

My statement is simply that it is redundant to: Enter Mob Destroy -> Enter Ghostize -> Attempt to clear mind -> Return to Mob Destroy -> Attempt to clear mind

What I am saying is Ghostize clearing mind ONLY for a qdel is redundant because we know we should already be in a Destroy call that will clear it. So whats the point of even checking if we're deleted if the Deletion proc will know it is deleted and can just clear it once and for all?